pyvrp/Model.py

```diff
@@ -273,12 +323,18 @@ def data(self) -> ProblemData:
             distances[frm, to] = edge.distance
             durations[frm, to] = edge.duration
 
+        members = [set(group.clients) for group in self._groups]
+        for idx, client in enumerate(self._clients, len(self._depots)):
+            if client.group is not None and idx not in members[client.group]:
+                self._groups[client.group].add_client(idx)
```

Comment:
The state of the ClientGroup is a bit confusing now: it only perfectly captures the client indices directly after calling `data()`. After calling `add_client` and `add_depot`, the client group's indices are wrong.

Since we are exposing client groups to users, we should ensure that the client group data is always correct.

Suggestion:

- The selected code can be refactored to be called `_update_groups`.
- When adding a client with a group, this client can be simply added to the corresponding group.
- When adding a depot, we clear all groups and then call `_update_groups` to populate them again with new indices. (we might not even need a separate `_update_groups` method)
- In `data()` we don't need to update the groups anymore.